### ISSUE TITLE
Small context independence bugs

### ISSUE THREAD

**Issue by cyrus-** [cyrus-/relit#10]
There are two issues in the code that is generated right now for a TLM application, e.g.:
```
let regex =
  /* open Pervasives to ensure context independence */
  Pervasives.(
    /* open $regex's Dependencies, again insuring context indepedence
       (RelitInternalDefn_regex is Relit's internal way of saying $regex) */
    Regex_notation.RelitInternalDefn_regex.Dependencies
      .(
      () => (
        Regex.Or(
          Regex.Str("a"),
          Regex.Star(Regex.Str("b")),
        /* a type assertion that the expansion is the type the notation definition specified */
        ): Regex_notation.RelitInternalDefn_regex.t
      )
    )
  )();
```

First, the module Regex_notation in the path to the dependencies will be resolved relative to Pervasives, which is incorrect. Second, the expansion type annotation will be resolved relative to the Dependencies, which is also incorrect. Very unlikely to encounter either bug, but we should fix this. I think the correct code that should be generated is:

```
let regex = {
  module RelitInternalDependencies = Regex_notation.RelitInternalDefn_regex.Dependencies;
  Pervasives.(
      RelitInternalDependencies
      .(
      () => (
        Regex.Or(
          Regex.Str("a"),
          Regex.Star(Regex.Str("b"))
        )
      )
    )
  )() : Regex_notation.RelitInternalDefn_regex.t;
};
```

**Comment by charlesetc** [comment]
Fixed :)

**Comment by cyrus-** [comment]
Opening Pervasives after Dependencies doesn't work. Then stuff in Pervasives will shadow dependencies. Need to give a unique name to Dependencies and then do it in the original order.

**Comment by charlesetc** [comment]
Whoops true... I think I fixed it with e669937cf0b1581c87422af3ff5b271954120641! Think of any ways that could go wrong?

**Comment by cyrus-** [comment]
Won't opening `Regex_notation.RelitInternalDefn.regex` bring all of the misc stuff in that module into scope, e.g. the `Apply` exception and the type `t`.

I think you really do need to do a `let module`, which is definitely a thing you can do in the OCaml AST. I assume Reason will pretty-print it with a block like in my initial post in the issue?

**Comment by charlesetc** [comment]
Yeah it will, but those names can't conflict with anything in pervasives, nor can the expanded code reference any of them, since we typecheck it separately. It's a strange solution, but it's simpler and I'm fairly confident it's correct.

**Comment by cyrus-** [comment]
Ah okay. Can you write a test that shows that you can't refer to type `t` accidentally in the expansion if its not in the dependencies?
